app/data_model/answer_store.py

```diff
@@ -51,13 +56,31 @@ def copy(self):
     def add_or_update(self, answer):
         """
         Add a new answer into the answer store, or update if it exists.
-
         :param answer: An answer object.
         """
         self._validate(answer)
+        position = self.find(answer)
 
-        answer_to_add = vars(answer).copy()
-        self.answer_map[answer.answer_id] = answer_to_add
+        if position is None:
+            answer_to_add = vars(answer).copy()
+            self.answer_map[answer_to_add['answer_id']].append(answer_to_add)
+        else:
+            self.answer_map[answer.answer_id][position]['value'] = answer.value
```

Comment:
Now we only have one identifier for something that repeats, we should consider using it as a key rather than position in a list as it would be more efficient for creating and looking up answers.

Reply:
I've implemented the naive approach for this here: https://github.com/ONSdigital/eq-survey-runner/tree/lists-new-answerstore

Discussed with @ajmaddaford, but not really happy with the implementation having to use two different types under each answer_id.

Also discussed some alternative structures for the answer_store, but the best approach seems to be to use a tuple of (`answer_id`, `list_item_id`) as a key for the answer store, and store a single answer under each key.

For non-repeating answers, this means we can query using (<answer_id>, None).

For repeating answers, if you have both answer_id and list_item_id, querying is efficient: (<answer_id>, <list_item_id>).

Where this solution is not so efficient is if you are querying a repeating answer with only a `list_item_id` or `answer_id`. I can't see any use cases where we would need that.

Currently, this will only be implemented for the in-memory structure for the answer_store. The serialisation format should be decided based on the most efficient serialisation / deserialisation routine.

This will mean implementing a custom serialise / deserialise method on the answer_store.

Reply:
Also discussed removing the filter method from the answer_store and replacing with methods to retrieve one answer and multiple answers.